### PR TITLE
Un-nest methods inside monomorph run

### PR DESCRIPTION
**Pull request by JonathanStarup**
Additionally removed unused variables. It is impossible to make the diff readable but now its self contained. It could be nice if the diff could be assisted somehow.

### CODE REVIEW

**Comment by magnus-madsen** [comment]
This only unnests right? Could you make all methods private?

**Comment by magnus-madsen** [comment]
If it only unnests I will just merge it, trusting that nothing breaks :)

File: main/src/ca/uwaterloo/flix/language/phase/Monomorph.scala [hunk 1]
```diff
@@ -124,683 +124,683 @@ object Monomorph extends Phase[TypedAst.Root, TypedAst.Root] {
...
-      * A function-local queue of pending (fresh symbol, function definition, and substitution)-triples.
-      *
-      * For example, if the queue contains the entry:
-      *
-      * -   (f$1, f, [a -> Int])
-      *
-      * it means that the function definition f should be specialized w.r.t. the map [a -> Int] under the fresh name f$1.
-      */
-    val defQueue: mutable.Queue[(Symbol.DefnSym, Def, StrictSubstitution)] = mutable.Queue.empty
+  private type DefQueue = mutable.Queue[(Symbol.DefnSym, Def, StrictSubstitution)]
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Monomorph.scala:135]: Wait, but this is not safe. If two Flix compilers are running they will use the same queue!

File: main/src/ca/uwaterloo/flix/language/phase/Monomorph.scala [hunk 2]
```diff
@@ -124,683 +124,683 @@ object Monomorph extends Phase[TypedAst.Root, TypedAst.Root] {
...
+    *
+    * For example, if the function:
+    *
+    * -   def fst[a, b](x: a, y: b): a = ...
+    *
+    * has been specialized w.r.t. to `Int` and `Str` then this map will contain an entry:
+    *
+    * -   (fst, (Int, Str) -> Int) -> fst$1
+    */
+  private type Def2Def = mutable.Map[(Symbol.DefnSym, Type), Symbol.DefnSym]
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Monomorph.scala:148]: Sorry, the above comment applies here. Not to the type.

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Monomorph.scala:148]: What can we do? Pass it as an argument? Then we also need to pass the queue? So two arguments?

Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/phase/Monomorph.scala:148]: Only the Def2Def/DefQueue types are declared in the object scope, each run makes its own two separate def2def and defQueue (157, 159) and passes them around.

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Monomorph.scala:148]: Ah yes, it was early and I am sleepy.

**Comment by magnus-madsen** [comment]
So this should be safe to merge?

**Pull request merged**